Common/src/matrix_structure.cpp

```diff
@@ -1425,17 +1425,12 @@ void CSysMatrix::BuildILUPreconditioner(bool transposed) {
            upper triangular part, then multiply and modify the matrix.
            Here, Aik' = Aik - Aij*inv(Ajj)*Ajk. ---*/
           
-          if (kPoint < (long)nPointDomain) {
+          if ((kPoint >= jPoint) && (jPoint < (long)nPointDomain)) {
+            
             Block_jk = GetBlock_ILUMatrix(jPoint, kPoint);
-            if (kPoint >= jPoint) {
-              
-              // WARNING: here we have a left multiply by Block_jk, should it
-              // be a right multiply to give Aik' = Aik - Aij*inv(Ajj)*Ajk?
-              
-              MatrixMatrixProduct(Block_jk, block_weight, block);
-              SubtractBlock_ILUMatrix(iPoint, kPoint, block);
-              
-            }
+            MatrixMatrixProduct(block_weight, Block_jk, block);
```

Comment:
Thanks for the fix for ILU. In your experience, is ILU performing much better than LU_SGS for both SU2_DEF and SU2_CFD too (flow problems)?

Reply:
Yes, ILU0 is working better (or much better) in both problems. Just try the following... RAE2822 turbulent simulation and
LINEAR_SOLVER_ERROR= 1E-10
LINEAR_SOLVER_ITER= 100

Now ILU requires ~20% less iterations than the LU_SGS to achieve the same level of convergence. Before, ILU was worst than LU_SGS... same situation with the grid deformation. 

Since ILU is working I have also changed my way to run SU2... Now I'm always running the solver with a CFL ~10 before it was ~1... the price is that I'm performing more LINEAR_SOLVER_ITER iterations but it is totally worth it and the CFL numbers are competitive with other solvers.

Now that I understand the ILU implementation I'm planning to work on a ILU1 implementation.